edward/inferences/bigan_inference.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import six
+import tensorflow as tf
+
+from edward.inferences.gan_inference import GANInference
+from edward.util import get_session
+
+
+class BiGANInference(GANInference):
+  """Adversarially Learned Inference (Dumoulin et al., 2017) or
+  Bidirectional Generative Adversarial Networks (Donahue et al., 2017)
+  for joint learning of generator and inference networks.
+
+  Works for the class of implicit (and differentiable) probabilistic
+  models. These models do not require a tractable density and assume
+  only a program that generates samples.
+  """
+  def __init__(self, latent_vars, data, discriminator):
+    """
+    Notes
+    -----
+    ``BiGANInference`` matches a mapping from data to latent variables and a
+    mapping from latent variables to data through a joint
+    discriminator.
+
+    In building the computation graph for inference, the
+    discriminator's parameters can be accessed with the variable scope
+    "Disc".
+    In building the computation graph for inference, the
+    encoder and decoder parameters can be accessed with the variable scope
+    "Gen".
+
+    Examples
+    --------
+    >>> with tf.variable_scope("Gen"):
+    >>>   xf = gen_data(z_ph)
+    >>>   zf = gen_latent(x_ph)
+    >>> inference = ed.BiGANInference({latent_vars: qz}, {xf: x_data}, discriminator)
```

Comment:
Should this line be `{z_ph: zf}` and `{xf: x_ph}`? Trying to map notation to the lines above.